### DISCUSSION TITLE
Changing extensions API to separate streaming output from output_modifier

### DISCUSSION THREAD

**Discussion by Wojtab** [oobabooga/text-generation-webui#1332] [Ideas]
I tried Silero TTS and was annoyed that streaming output isn't shown live, so I took a look into extensions, and I seen that both SD, and TTS disable streaming during generation, as it would break them.

To fix it, I came up with an idea, to create a new hook, called for example `stream_modifier`, which would get called instead of `output_modifier` when streaming, and only when the generation has finished the `output_modifier` would get called. I implemented a quick&dirty POC, and added it to Silero, however the full HTML reload of the entire history on every token looked bad (for example the audio players were flashing), so I decided to separate the current message from chat history. Right now, it looks like that:

https://user-images.githubusercontent.com/3718215/232648179-4a91bfac-4864-409a-ba12-e30c2cfbc0f4.mp4

You can try it out, by applying the following patch (on commit `ac2973f`): [streaming.patch](https://github.com/oobabooga/text-generation-webui/files/11257140/streaming.patch) and then going to cai-chat (I broke the other chat modes), with just silero-tts enabled (I didn't test other extensions).

If this idea gains support, and my way of implementing it isn't completly insane, I will bring it to PR stage when I have some time.

**Comment by MotherSoraka** [comment]
Why is this not the default procedure?
and why hasnt this post hasnt gained any support?
:|
